cosmos/providers/dbt/core/operators/kubernetes.py

```diff
@@ -0,0 +1,176 @@
+        self.build_env_args(env_vars)
+        self.arguments = dbt_cmd
+
+
+class DbtLSKubernetesOperator(DbtKubernetesBaseOperator):
+    """
+    Executes a dbt core ls command.
+    """
+
+    ui_color = "#DBCDF6"
+
+    def __init__(self, **kwargs) -> None:
+        super().__init__(**kwargs)
+        self.base_cmd = "ls"
+
+    def execute(self, context: Context):
+        return self.build_and_run_cmd(context=context)
+
+
+class DbtSeedKubernetesOperator(DbtKubernetesBaseOperator):
+    """
+    Executes a dbt core seed command.
+
+    :param full_refresh: dbt optional arg - dbt will treat incremental models as table models
+    """
+
+    ui_color = "#F58D7E"
+
+    def __init__(self, full_refresh: bool = False, **kwargs) -> None:
+        self.full_refresh = full_refresh
+        super().__init__(**kwargs)
+        self.base_cmd = "seed"
+
+    def add_cmd_flags(self):
+        flags = []
+        if self.full_refresh is True:
+            flags.append("--full-refresh")
+
+        return flags
+
+    def execute(self, context: Context):
+        cmd_flags = self.add_cmd_flags()
+        return self.build_and_run_cmd(context=context, cmd_flags=cmd_flags)
+
+
+class DbtRunKubernetesOperator(DbtKubernetesBaseOperator):
+    """
+    Executes a dbt core run command.
+    """
+
+    ui_color = "#7352BA"
+    ui_fgcolor = "#F4F2FC"
+
+    def __init__(self, **kwargs) -> None:
+        super().__init__(**kwargs)
+        self.base_cmd = "run"
+
+    def execute(self, context: Context):
+        return self.build_and_run_cmd(context=context)
+
+
+class DbtTestKubernetesOperator(DbtKubernetesBaseOperator):
+    """
+    Executes a dbt core test command.
+    """
+
+    ui_color = "#8194E0"
+
+    def __init__(self, **kwargs) -> None:
+        super().__init__(**kwargs)
+        self.base_cmd = "test"
+
+    def execute(self, context: Context):
+        return self.build_and_run_cmd(context=context)
+
+
+class DbtRunOperationKubernetesOperator(DbtKubernetesBaseOperator):
+    """
+    Executes a dbt core run-operation command.
+
+    :param macro_name: name of macro to execute
+    :param args: Supply arguments to the macro. This dictionary will be mapped to the keyword arguments defined in the
+        selected macro.
+    """
+
+    ui_color = "#8194E0"
+    template_fields: Sequence[str] = "args"
+
+    def __init__(self, macro_name: str, args: dict = None, **kwargs) -> None:
+        self.macro_name = macro_name
+        self.args = args
+        super().__init__(**kwargs)
+        self.base_cmd = ["run-operation", macro_name]
+
+    def add_cmd_flags(self):
+        flags = []
+        if self.args is not None:
+            flags.append("--args")
+            flags.append(yaml.dump(self.args))
+        return flags
+
+    def execute(self, context: Context):
+        cmd_flags = self.add_cmd_flags()
+        return self.build_and_run_cmd(context=context, cmd_flags=cmd_flags)
+
+
+class DbtDepsKubernetesOperator(DbtKubernetesBaseOperator):
```

Comment:
@juldrixx does the `DepsOperator` make sense in the case of `Docker` and `Kubernetes`? Perhaps it would make more sense to have this as an argument of the `Run` and `Seed` operators, WDYT?

Reply:
I see what you mean, it's true that doing it in isolation has no use. In our case we execute it during the construction of the Docker image because we have the dbt project in it. That's why I didn't pay attention to it and just copied the logic of the local way.

And in some cases (mine for example), if we install packages in a container at runtime, it triggers a security alert so maybe it should be optional.

but it is certain that if we inject the dbt project into the container, we will have to install these dependencies at runtime. And maybe if you don't want to do it you should provide your own image with them in it.

Reply:
Even in the context of Airflow, the Deps operator should be removed. It would not work if the Deps task runs in a different Airflow worker from where the Run and Seed operators are being run...! But this is a matter for another PR.

Suggestions for the current PR:
1. Remove `DbtDepsKubernetesOperator` and `DbtDepsDockerOperator`
2. We'll need to add to the documentation that `deps` should be handled within the container image-building process

Reply:
It's done, I also added the re-export for the `DbtDepsLocalOperator` because it wasn't present for some reason

Reply:
Thank you very much!

Reply:
BTW: just as a reference, I logged this ticket: #218